plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2053,6 +2077,36 @@ static auto maybe_make_debug_time_logger() -> std::optional<decltype(make_debug_
    }
 }
 
+void block_only_sync::schedule() {
+   if (!_pending) {
+      // wait one second to see if we can actually get the block from net plugin before we try to resync from block vault
+      _start_sync_timer.expires_from_now(boost::posix_time::seconds(1));
+      _pending = true;
+      _start_sync_timer.async_wait(
+          app().get_priority_queue().wrap(priority::high, [this](const boost::system::error_code& ec) {
+             if (ec != boost::asio::error::operation_aborted) {
+                auto id = _impl->chain_plug->chain().last_irreversible_block_id();
+                fc_dlog(_log, "Attempt to resync from block vault");
+                _impl->blockvault->sync(&id, *this);
```

Comment:
this can throw (on a snapshot or a few different block errors) which will bubble from here out.   We should ensure that we don't throw from a job handler so something here needs to catch errors and respond to them appropriately